utils/nxdl2rst.py

```diff
@@ -671,19 +666,18 @@ def print_rst_from_nxdl(nxdl_file):
     lexical_name = name[2:]  # without padding 'NX', for indexing
 
     # retrieve category from directory
-    # subdir = os.path.split(os.path.split(tree.docinfo.URL)[0])[1]
-    subdir = root.attrib["category"]
+    category = root.attrib["category"]
 
     # Pass these terms to construct the full URL
     anchor_registry.nxdl_file = nxdl_file
-    anchor_registry.category = SUBDIR_MAP[subdir]
+    anchor_registry.subdir = subdir
 
     # TODO: check for consistency with root.get('category')
     listing_category = {
         "base": "base class",
         "application": "application definition",
         "contributed": "contributed definition",
-    }[subdir]
+    }[category]
```

Comment:
This is wrong I think. The `category` will never be `contributed`. So you'll have to use the `subdir` here.

Reply:
This is what I did: https://github.com/nexusformat/definitions/pull/1151

```python
listing_category = {
        "base_classes": "base class",
        "applications": "application definition",
        "contributed_definitions": "contributed definition",
    }[subdir]
```